easybuild/tools/filetools.py

```diff
@@ -392,8 +400,8 @@ def extract_file(fn, dest, cmd=None, extra_options=None, overwrite=False, forced
     abs_dest = os.path.abspath(dest)
 
     # change working directory
-    _log.debug("Unpacking %s in directory %s.", fn, abs_dest)
-    change_dir(abs_dest)
+    _log.debug("Unpacking %s in directory %s", fn, abs_dest)
+    cwd = change_dir(abs_dest)
```

Comment:
There is a problem here from test_guess_start_dir which does:
```
        # default (no start_dir specified): use unpacked dir as start dir
        self.assertEqual(ec['ec']['start_dir'], None)
        check_start_dir('toy-0.0')

        # using start_dir equal to the one we're in is OK
        ec['ec']['start_dir'] = '%(name)s-%(version)s'
        self.assertEqual(ec['ec']['start_dir'], 'toy-0.0')
        check_start_dir('toy-0.0')
```
That second check_start_dir('toy-0.0') has cwd %(builddir)s/toy-0.0 when it starts, the run_all_steps kicks off
ready_steps which is make_builddir which removes the old builddir thus causing the change_dir(abs_dest) to return None since builddir no longer exists.

Solution would perhaps be to compare abs_dest and cwd before that change_dir to see if they are the same, or if cwd is below abs_dest, and then set cwd to abs_dest

Reply:
This is just an issue in the test, easy fix is to change back to where the test started from on each call to `check_start_dir`, done in 3a4bc9860